src/cdk/custom-resources/upload-public-ssh/lambda/index.ts

```diff
@@ -7,7 +7,28 @@ import { IAM } from 'aws-sdk'
  */
 const iam = new IAM()
 
-export const handler = async (event: any): Promise<void> => {
+/**
+ * See the AWS documentation for more information passed in the request for a custom resource.
+ *
+ * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+ */
+export interface CustomResourceProviderRequest<T1, T2> {
+  RequestType: 'Create' | 'Update' | 'Delete'
+  ResponseURL: string
+  StackId: string
+  RequestId: string
+  ResourceType: string
+  LogicalResourceId: string
+  ResourceProperties: T1
+  OldResourceProperties: T2
+}
+
+interface Properties {
+  UserName: string
+  SSHPublicKeyBody: string
+}
+
+export const handler = async (event: CustomResourceProviderRequest<Properties, Properties>): Promise<void> => {
```

Comment:
It seems like since lambda is invoked by the AWS service user, this typing won't actually get parsed by TSC (because outside of tests it's never explicitly called)?